tiling.js

```diff
@@ -2581,6 +2581,82 @@ var Spaces = class Spaces extends Map {
 
 Signals.addSignalMethods(Spaces.prototype);
 
+/**
+ * Return true if a window is tiled (e.g. not floating, not scratch, not transient).
+ * @param metaWindow 
+ */
+function isTiled(metaWindow) {
+    if (!metaWindow) {
+        return false;
+    }
+
+    if (!isFloating(metaWindow) && 
+        !isScratch(metaWindow) && 
+        !isTransient(metaWindow)) {
+        return true;
+    }
+    else {
+        return false;
+    }
+}
+
+/**
+ * Transient windows are connected to a parent window and take focus.
+ * On Wayland it takes entire focus (can't focus parent window while it's open).
+ * @param metaWindow 
+ * @returns 
+ */
+function isTransient(metaWindow) {
+    if (!metaWindow) {
+        return false;
+    }
+    if (metaWindow.get_transient_for()) {
+        return true;
+    }
+    else {
+        return false;
+    }
+}
+
+/**
+ * Returns true iff a metaWindows has at least one transient window.
```

Comment:
Oh, that was habit there - "iff" stands for "if and only if".  It usually refers to bidirectional conditional statements, and my mind sometimes does this to imply to the developer, or myself, that this method returns true "only if" it has transient windows... false otherwise (which you might think is implied, but from a strict logic point of view "iff" is intended to convey there's no other situations where this method return true).